CMGTools/H2TauTau/python/proto/analyzers/H2TauTauTreeProducer.py

```diff
@@ -0,0 +1,434 @@
+    def bookParticle( self, tree, pName ):
+        self.var(tree, '{pName}_pt'    .format(pName=pName))
+        self.var(tree, '{pName}_eta'   .format(pName=pName))
+        self.var(tree, '{pName}_phi'   .format(pName=pName))
+        self.var(tree, '{pName}_charge'.format(pName=pName))
+        self.var(tree, '{pName}_mass'  .format(pName=pName))
+    
+    def fillParticle( self, tree, pName, particle ):
+        self.fill(tree, '{pName}_pt'    .format(pName=pName), particle.pt()     )
+        self.fill(tree, '{pName}_eta'   .format(pName=pName), particle.eta()    )
+        self.fill(tree, '{pName}_phi'   .format(pName=pName), particle.phi()    )
+        self.fill(tree, '{pName}_charge'.format(pName=pName), particle.charge() )
+        self.fill(tree, '{pName}_mass'  .format(pName=pName), particle.mass()   )
+    
+    
+    # simple gen particle
+    def bookGenParticle( self, tree, pName ):
+        self.bookParticle(tree, pName)
+        self.var(tree, '{pName}_mass' .format(pName=pName))
+        self.var(tree, '{pName}_pdgId'.format(pName=pName))
+        
+    def fillGenParticle( self, tree, pName, particle ):
+        self.fillParticle(tree, pName, particle)
+        self.fill(tree, '{pName}_mass' .format(pName=pName), particle.mass() )
+        self.fill(tree, '{pName}_pdgId'.format(pName=pName), particle.pdgId())
+    
+    
+    # di-tau 
+    def bookDiLepton( self, tree ):
+        self.var( tree, 'visMass'      )
+        self.var( tree, 'svfitMass'    )
+        self.var( tree, 'pZetaMET'     )
+        self.var( tree, 'pZetaVis'     )
+        self.var( tree, 'pZetaDisc'    )
+        self.var( tree, 'mtleg2'       )
+        self.var( tree, 'mtleg1'       )
+        self.var( tree, 'mt'           )
+        self.var( tree, 'pthiggs'      )
+        self.var( tree, 'deltaPhiL1L2' )
+        self.var( tree, 'deltaEtaL1L2' )
+        self.var( tree, 'deltaRL1L2'   )
+        self.var( tree, 'deltaPhiL1MET')
+        self.var( tree, 'deltaPhiL2MET')
+        self.var( tree, 'metcov00'     )
+        self.var( tree, 'metcov01'     )
+        self.var( tree, 'metcov10'     )
+        self.var( tree, 'metcov11'     )
+        self.var( tree, 'metphi'       )
+        self.var( tree, 'mex'          )
+        self.var( tree, 'mey'          )
+        self.var( tree, 'met'          )
+    
+    def fillDiLepton( self, tree, diLepton):
+        self.fill(tree, 'visMass'  , diLepton.mass()     )
+        self.fill(tree, 'svfitMass', diLepton.svfitMass())
+        self.fill(tree, 'pZetaMET' , diLepton.pZetaMET() )
+        self.fill(tree, 'pZetaVis' , diLepton.pZetaVis() )
+        self.fill(tree, 'pZetaDisc', diLepton.pZetaDisc())
+        self.fill(tree, 'mt'       , diLepton.mTLeg2()   )
+        self.fill(tree, 'mtleg2'   , diLepton.mTLeg2()   ) # RIC: handy for tt, redundant but non destructive
+        self.fill(tree, 'mtleg1'   , diLepton.mTLeg1()   )
+        # RIC: can't find cov matrix... will look it up later
```

Comment:
diLepton.met().getSignificanceMatrix() crashes.
Agreed to go ahead without this is and take not that this needs to be fixed.